### ISSUE TITLE
Option to disable moving mouse cursor to clicked item

### ISSUE THREAD

**Issue by benwoodward** [dexterleng/vimac#412]
It seems like Vimac moving the mouse cursor to whatever was clicked is new, I don't remember it doing that before. It'd be great if it was optional, it'd be good if I could click a button without having to move my mouse cursor out of the way (which often happens when I'm trying to read something).

**Comment by blackketter** [comment]
There's a related discussion in #251

Is the issue that the mouse pointer has been moved, or that it's visible?  It seems to me that the mouse could be kept hidden (typically in macOS the mouse is hidden after typing in text) but still be moved to the location of the click.

**Comment by benwoodward** [comment]
For me the issue is that the mouse cursor is moved. When I'm reading I keep it off to one side of the text. If the mouse cursor is move to where the text is because I clicked a link in the text then I have to move it back. However, if it just remained hidden this wouldn't be an issue.

**Comment by dexterleng** [comment]
"revert cursor to original position" was removed because of #251, but I've since added the 'move operation' that solely moves the cursor without clicking. I think "revert cursor" can be added back? @kiding What are your thoughts?

**Comment by kiding** [comment]
Yes, it can be added back. Although I’d still prefer an option in the preferences, I suspect the majority of users would like the cursor to be stationary.

**Comment by benwoodward** [comment]
Makes sense to me now why it needs to be moved. Where can I read about the 'move operation'?

**Comment by dexterleng** [comment]
> Yes, it can be added back. Although I’d still prefer an option in the preferences, I suspect the majority of users would like the cursor to be stationary.

Yeah, I think aesthetically speaking keeping it in the same original position would be preferred. But is there a reason this should be configurable now that you can execute a move operation?

**Comment by dexterleng** [comment]
I think I'll add an experimental option for this.

**Comment by blackketter** [comment]
For me, the issue is that the mouse remain hidden after a Vimac click.  I don't mind that it's moved to where the click happens, that seems natural.

**Comment by dexterleng** [comment]
> For me, the issue is that the mouse remain hidden after a Vimac click

Could you elaborate on this? On the latest version v0.3.18 the mouse should reappear at the location a click is performed.

**Comment by blackketter** [comment]
> Could you elaborate on this? On the latest version v0.3.18 the mouse should reappear at the location a click is performed.

My preference would be that the mouse pointer should be hidden by default in Vimac.  The _location_ of that hidden pointer should probably be moved to where a click is generated by Vimac.  I find the appearance of the cursor a distraction.

It might make sense to provide some kind of short-term feedback when a click happens.  That might be a short animation showing the click.  (Like screen recorder does, maybe?)

https://user-images.githubusercontent.com/131679/115761385-2305d880-a357-11eb-9299-cada84193c92.mov

When using the new move option, though, it would make sense to show the cursor in the new location, as that's the feedback that the move worked.

**Comment by dexterleng** [comment]
> My preference would be that the mouse pointer should be hidden by default in Vimac

So it moves the cursor to the target location, keeps it there until the user disturbs it by moving their mouse?

**Comment by blackketter** [comment]
> So Vimac moves the cursor to the target location, performs a click, keeps it at location hidden, until the user disturbs it by moving their mouse?

Exactly!
